@here/harp-map-controls/lib/MapControls.ts

```diff
@@ -463,21 +463,37 @@ export class MapControls extends THREE.EventDispatcher {
         }
 
         this.dispatchEvent(MAPCONTROL_EVENT_BEGIN_INTERACTION);
-
         // Register the zoom request
         this.m_startZoom = this.currentZoom;
         this.m_zoomDeltaRequested = zoomLevel - this.zoomLevelTargetted;
-
         // Cancel panning so the point of origin of the zoom is maintained.
         this.m_panDistanceFrameDelta.set(0, 0, 0);
         this.m_lastAveragedPanDistance = 0;
 
-        this.m_zoomTargetNormalizedCoordinates.set(screenTarget.x, screenTarget.y);
-
         // Assign the new animation start time.
         this.m_zoomAnimationStartTime = performance.now();
 
-        this.handleZoom();
+        if (this.mapView.projection.type === geoUtils.ProjectionType.Planar) {
+            this.m_zoomTargetNormalizedCoordinates.set(screenTarget.x, screenTarget.y);
+            this.handleZoom();
+        } else {
+            const L = this.mapView.projection.groundDistance(this.camera.position);
+
+            const surfaceNormal = this.mapView.projection.surfaceNormal(
+                this.camera.position,
+                new THREE.Vector3()
+            );
+
+            this.camera.position.addScaledVector(
+                surfaceNormal,
+                (this.zoomLevelTargetted - zoomLevel) * L * 0.1
+            );
+
+            if (this.mapView.projection.groundDistance(this.camera.position) < 500) {
```

Comment:
Don't know... copied from the GlobeControls, didn't do functional change. @robertoraggi ? Was that maybe a remnant of an early version of the globe?

Reply:
I guess the values of the min distance to the surface and the max distance from the surface should be configurable because they depend from the data (e.g. the max storage level of the data source), the height provider, and so on. 500 is the number that look ok to avoid precision issues when rendering data at storage level 17.

Reply:
I will add a comment that this limitation should be configurable.